### PR TITLE
Add PreserveSeparator flag to the carbon plugin

### PR DESCRIPTION
**Pull request by joekor**

### CODE REVIEW

**Comment by traylenator** [comment]
This appears to fix the beaker tests also.

File: manifests/plugin/write_graphite/carbon.pp [hunk 1]
```diff
@@ -12,6 +12,7 @@
   $protocol                  = 'tcp',
   Boolean $separateinstances = false,
   Boolean $logsenderrors     = true,
+  Boolean $preserveseparator = false,
```
Comment (bastelfreak) [inline manifests/plugin/write_graphite/carbon.pp]: Datatypes \o/

File: templates/plugin/write_graphite/carbon.conf.erb [hunk 2]
```diff
@@ -17,6 +17,9 @@
   LogSendErrors <%= @logsenderrors %>
   Protocol "<%= @protocol %>"
 <%- end -%>
+<%- if scope.lookupvar('collectd::collectd_version_real') and (scope.function_versioncmp([scope.lookupvar('collectd::collectd_version_real'), '5.7']) >= 0) -%>
+  PreserveSeparator <%= @preserveseparator %>
```
Comment (bastelfreak) [inline templates/plugin/write_graphite/carbon.conf.erb]: Does that work or do we need to convert it do a string? Please add a unit or acceptance test for this that verifies the file content

File: templates/plugin/write_graphite/carbon.conf.erb [hunk 3]
```diff
@@ -17,6 +17,9 @@
   LogSendErrors <%= @logsenderrors %>
   Protocol "<%= @protocol %>"
 <%- end -%>
+<%- if scope.lookupvar('collectd::collectd_version_real') and (scope.function_versioncmp([scope.lookupvar('collectd::collectd_version_real'), '5.7']) >= 0) -%>
```
Comment (bastelfreak) [inline templates/plugin/write_graphite/carbon.conf.erb]: scope.lookupvar() shoudn't be needed anymore.

**Comment by bastelfreak** [comment]
@joekor can you please take a look at the inline comments?

**Comment by dmorda** [comment]
Any chance this one will get merged soon? I've got collectd completely puppetized, but this one parameter is a deal breaker for me getting it working with Grafana.

**Comment by joekor** [comment]
I had totally forgotten about it - i’ll have a look tomorrow.


On 14 Oct 2018, at 12:26, Tim Meusel <notifications@github.com<mailto:notifications@github.com>> wrote:


@dmorda<https://github.com/dmorda> as you can see in the inline comments, this code requires some work. You can clone it locally, fix it and resubmit it as a new pull request if you want.

**Comment by bastelfreak** [comment]
Hi. I'm going to close this PR ddue to inactivity. Please reopen and rebase it if you're still interested in it.

**Pull request closed without merging**